### ISSUE TITLE
Sector query by id not working

### ISSUE THREAD

**Issue by thomasstjerne** [CatalogueOfLife/backend#1126]
I think this used work as a search, but the result is empty:
https://api.checklistbank.org/dataset/3/sector?id=489
However, this query gives exactly one sector with id 489:
https://api.checklistbank.org/dataset/3/sector?datasetKey=3&limit=100&offset=0&subjectDatasetKey=1172

**Comment by mdoering** [comment]
https://api.checklistbank.org/dataset/3/sector/489 ?

**Comment by thomasstjerne** [comment]
Yes, know that one, but it is in use on a page that has a table (result set) and takes various query params. So it was convenient to use the same endpoint for all queries.
It looks like the param does _something_ as the result is empty. If I do https://api.checklistbank.org/dataset/3/sector?nonsens=489 the param is ignored.
If it is removed, it should be removed in the API doc as well

**Comment by mdoering** [comment]
but it compares the subject or target id. Not sure if that changed at some point?

`AND (s.subject_id = #{req.id} OR s.target_id = #{req.id})`

**Comment by mdoering** [comment]
That makes sense though as you can get a single sector by its id via the regular REST resource

**Comment by mdoering** [comment]
it's the same for the decision API which does `AND ed.subject_id = #{req.id}`

**Comment by mdoering** [comment]
surely badly documented!
